### ISSUE TITLE
Very long learning process

### ISSUE THREAD

**Issue by fast8wild** [Unity-Technologies/ml-agents#253]
So I am currently developing an AI for a puyo-puyo (puzzle game) clone I made, and so far I have been training it for about 37 hours and have seen minimal improvement. I am not sure if it is because of the complexity of the given problem, the randomness of the piece pool at the start of each episode, or the values I put in the hyperparameters.  Any feedback would be appreciated.
![stats](https://user-images.githubusercontent.com/35200087/34974247-1711556a-fa9d-11e7-80ef-ca71db3545ec.png)

**Comment by MarcoMeter** [comment]
Hi @fast8wild 
there is by far not enough information provided to enable people to help you out.

What are your inputs, output actions and reward signals?

**Comment by fast8wild** [comment]
Inputs are continuous:
I needed to provide it details of both his and the opponents grid as well as each player's upcoming pieces. There are 4 color possibilities, 5 including garbage, so I implemented one hot encoding to represent each color.
Since each grid is 12 by 6, and there are 3 current/upcoming pieces (each holding 2 colors), the total state size is 780.

Outputs are discrete with an action size of 5:
Move left,
Move down,
Move right
Rotate Counterclockwise
Rotate Clockwise

Rewards are given when either
a) player makes a chain (chainNo/10).
b) player fills their screen (winner( gets 1.0, loser gets-1.0).

I should specify I am dealing with an external brain and a heuristic brain that are competing with one another.

**Comment by MarcoMeter** [comment]
Sounds like quite a huge state space. Now it would be interesting to see your hyperparameters.

Right now I'd suggest to scale down your environment, like using a smaller grid. As soon as good training results kick in, you can increase the complexity.

**Comment by fast8wild** [comment]
Using a smaller grid might be worth looking into. Though I am not sure of how I will be able to implement that without constantly changing the unity environment .

Also here are the hyperparameters:

gamma | 0.99
lambd | 0.95
time_horizon | 2048
beta | 0.001
num_epoch | 5
num_layers | 3
epsilon | 0.2
buffe_size | 256000
leaning_rate | 0.0003
hidden_units | 512
batch_size | 256

**Comment by fast8wild** [comment]
> Right now I'd suggest to scale down your environment, like using a smaller grid. As soon as good training results kick in, you can increase the complexity.

So in that case, should I try gradually increase the grid width using a curriculum, or just do it manually since I see no way of changing the state size through a reset parameter.

**Comment by MarcoMeter** [comment]
You could use curriculum learning.

In the end you have to implement a dynamic solution for your environment so that you can easily make modifications to your environment. Just keep in mind that you cannot dynamically increase the state space. Once modifications are done to the neural network topology-wise, it has to be trained from scratch again.

**Comment by fast8wild** [comment]
So I guess in this case the best thing to do is keep the state size the same, but just limit the field of activity until the AI shows improvement.

**Comment by unityjeffrey** [comment]
Thanks for reaching out to us. Hopefully you were able to resolve your issue.  We are closing this due to inactivity.
